cmd/lookout/serve.go

```diff
@@ -230,6 +232,48 @@ func (c *ServeCommand) startServer(srv *lookout.DataServerHandler) error {
 	return nil
 }
 
+func (c *ServeCommand) initDB() (*sql.DB, error) {
+	db, err := sql.Open("postgres", c.DB)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = db.Ping(); err != nil {
+		return nil, err
+	}
+
+	m, err := store.NewMigrateInstance(db)
+	if err != nil {
+		return nil, err
+	}
+
+	dbVersion, _, err := m.Version()
+
+	// The DB is not initialized
+	if err == migrate.ErrNilVersion {
+		return nil, fmt.Errorf("the DB is empty, it needs to be initialized with the 'lookout migrate' command")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	maxVersion, err := store.MaxMigrateVersion()
+	if err != nil {
+		return nil, err
+	}
+
+	if dbVersion != maxVersion {
+		return nil, fmt.Errorf(
+			"database version mismatch. Current version is %v, but this binary needs version %v. "+
+				"Use 'lookout migrate' to upgrade your database", dbVersion, maxVersion)
```

Comment:
A nitpick, but would not referring to a specific binary version improve the ability to debug this failure?

E.g  `"Current DB version is %v, but this binary %v build %v needs DB version %v. ", ... version, build, ...`

Reply:
good idea, I'll squash this small change and merge